File: tests/csharp/tests-csharp-xunit-api_request-playwright/Tests/E2e/DeleteAccountTests.cs
using Allure.Net.Commons;
using Allure.Net.Commons.Attributes;
using Helpers;
using Tests;

namespace Tests.E2e;

[AllureLabel("layer", "e2e")]
[AllureEpic("Authentication")]
[AllureFeature("Delete account")]
[AllureSeverity(SeverityLevel.critical)]
[AllureSuite("Delete account")]
public sealed class DeleteAccountTests : TestBase
{
    [Trait("TestCategory", "e2e")]
    [Trait("TestCategory", "positive")]
    [Fact(DisplayName = "User can delete the account from home")]
    public void ShouldDeleteAccount()
    {
        var user = new UserBuilder().WithUsername().WithPassword().Build();
        RegisterPage.OpenPage().Signup(user.Username, user.Password);
        HomePage.ShouldHaveWelcomeMessage(user.WelcomeMessage())
            .ClickDeleteAccountAndConfirm()
            .ShouldHaveFormTitle("Login Form");
    }

    [Trait("TestCategory", "e2e")]
    [Fact(DisplayName = "Cancelling the confirm keeps the session")]
    public void CancellingConfirmKeepsSession()
    {
        var user = new UserBuilder().WithUsername().WithPassword().Build();
        RegisterPage.OpenPage().Signup(user.Username, user.Password);
        HomePage.ShouldHaveWelcomeMessage(user.WelcomeMessage())
            .ClickDeleteAccountAndCancel()
            .ShouldHaveWelcomeMessage(user.WelcomeMessage())
            .ShouldKeepAuthToken()
            .ClickDeleteAccountAndConfirm()
            .ShouldHaveFormTitle("Login Form");
    }
}

File: tests/java/tests-java-junit5-rest_assured-selenide-appium/src/test/java/tests/e2e/DeleteAccountTests.java
package tests.e2e;

import annotations.Layer;
import helpers.AuthSetup;
import helpers.DataFaker;
import io.qameta.allure.Epic;
import io.qameta.allure.Feature;
import io.qameta.allure.Severity;
import io.qameta.allure.SeverityLevel;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import tests.TestBase;

@Layer("e2e")
@Epic("Authentication")
@Feature("Delete account")
@Severity(SeverityLevel.CRITICAL)
@DisplayName("Delete account")
class DeleteAccountTests extends TestBase {

    private static final String PASSWORD = "password123";

    /** Throwaway — never seeded user1. Deleted through the API if the UI did not. */
    private String throwawayUsername;

    @AfterEach
    void cleanupThrowawayUser() {
        if (throwawayUsername != null) {
            AuthSetup.deleteAccountQuietly(throwawayUsername, PASSWORD);
            throwawayUsername = null;
        }
    }

    @Test
    @Tag("e2e")
    @Tag("positive")
    @DisplayName("Confirming delete account clears the session and navigates to login")
    void confirmingDeleteClearsSessionAndNavigatesToLogin() {
        throwawayUsername = DataFaker.username();
        AuthSetup.register(throwawayUsername, PASSWORD);

        loginScreen.shouldBeOpen()
                .fillAndSubmitForm(throwawayUsername, PASSWORD)
                .shouldHaveWelcomeMessage("Welcome, " + throwawayUsername + "!")
                .shouldShowSessionActions()
                .clickDeleteAccountAndConfirm()
                .shouldHaveFormTitle("Login Form");

        throwawayUsername = null;
    }

    @Test
    @Tag("e2e")
    @Tag("positive")
    @DisplayName("Cancelling the confirm keeps the session")
    void cancellingConfirmKeepsSession() {
        throwawayUsername = DataFaker.username();
        AuthSetup.register(throwawayUsername, PASSWORD);

        loginScreen.shouldBeOpen()
                .fillAndSubmitForm(throwawayUsername, PASSWORD)
                .shouldHaveWelcomeMessage("Welcome, " + throwawayUsername + "!")
                .clickDeleteAccountAndCancel()
                .shouldHaveWelcomeMessage("Welcome, " + throwawayUsername + "!");

        AuthSetup.login(throwawayUsername, PASSWORD);
    }
}
